src/v2/Utils/auth.ts

```diff
@@ -170,3 +173,41 @@ export const logout = async () => {
   const err = await response.json()
   return Promise.reject(new Error(err.error))
 }
+
+export enum Status {
+  LoggedIn,
+  LoggedOut,
+  Invalid,
+}
+
+/**
+ * Ensures authentication is still valid.
+ *
+ * If user is logged out; `Promise` resolves to `Status.LoggedOut`
+ * If user is logged in; `Promise` resolves to `Status.LoggedIn`
+ * If user is logged in with invalid authentication (401); `Promise` resolves to `Status.Invalid`
+ * All other status codes will also resolve to `Status.LoggedIn` because we don't know
+ * whether or not the authentication is valid (error could be something else)
+ */
+export const validate = async (): Promise<Status> => {
+  const currentUser = getENV("CURRENT_USER")
+
+  if (!currentUser) return Status.LoggedOut
+
+  const response = await fetch(`${getENV("API_URL")}/api/v1/me`, {
```

Comment:
To simplify this, can we just go through MP? 
```tsx
const me = await fetchQuery(graphql`
  query MeQuery {
    me {
      ...
    }
  }
`)

if (!me) {
  return Status.Invalid
}
```

Reply:
Yeah, good call

Reply:
Actually, in this case I don't think we're returning the status code. We could wrap up a proper check for this in Metaphysics though.

Reply:
Would be worth it.